### PR TITLE
Automatically download and cache atomic data files.

### PR DESCRIPTION
**Pull request by karandesai-96**
This PR makes changes with mechanism of making use of atomic dataset. Initially an atomic dataset had to be kept separately and provided through command line, but with the merge of this PR, it will be automatically downloaded, cached and used. The command line argument `--atomic-dataset` has been replaced by a flag `--remote-data`.

- [X] A session scoped fixture named `_atom_data` is made in top level **conftest.py**, which will download and cache atom data (currently only `kurucz_cd23_chianti_H_He.h5.zip`).
- [X] A fixture named `atom_data` is made alongwith it, to return a copy of this `AtomData` object returned by `_atom_data` fixture, to avoid re prepare.
- [X] Old fixtures in **conftest.py** have been removed and other tests have been modified to use this structure henceforth.
- [X] Suitable changes have been made in **.travis.yml** as well. Manually `wget` and `unzip` commands of atomic dataset file have been removed.

[Latest Report](http://opensupernova.org/~karandesai96/integration/doku.php?id=reports:c0cf29d)

### CODE REVIEW

**Comment by yeganer** [comment]
Please do the following:

- Create a fixture `_atom_data` inside tardis/conftest.py returning the cached atom_data
- Create a fixture `atom_data` returning `copy.copy(_atom_data)`
- Rewrite ALL tests so they use the top level atom_data fixture

**Comment by wkerzendorf** [comment]
@karandesai-96 rebase this branch

**Comment by karandesai-96** [comment]
@wkerzendorf Rebased.
@yeganer Directory name was changed `tests_slow` -> `integration_tests` hence overagerc didn't work. Rectified that as well.

File: tardis/tests/integration_tests/test_integration.py [hunk 1]
```diff
@@ -28,6 +30,7 @@ def setup(self, request, reference, data_path, atomic_data_fname):
         self.name = data_path['setup_name']
 
         self.config_file = os.path.join(data_path['config_dirpath'], "config.yml")
+        self.config_dict = yaml.load(self.config_file)
```
Comment (yeganer) [inline tardis/tests/integration_tests/test_integration.py]: That's bad, don't do this. We modify the `YAMLLoader`.
Use `from tardis import load_yaml` instead.

Comment (yeganer) [inline tardis/tests/integration_tests/test_integration.py]: Why did you change it anyway? To me it seems like `Configuration.from_yaml(...)` does everything we want.

We might even go so far as to use `tardis.run_tardis(config)`. Any reason to not use this?

File: tardis/conftest.py [hunk 2]
```diff
@@ -110,7 +92,40 @@ def included_he_atomic_data(test_data_path):
...
+    request.addfinalizer(fin)
+    return atom_data
+
+
+@pytest.fixture(scope="class")
+def atom_data(_atom_data):
+    if _atom_data.md5 != '21095dd25faa1683f4c90c911a00c3f8':
+        pytest.skip('Need default Kurucz atomic dataset '
+                    '(md5="21095dd25faa1683f4c90c911a00c3f8"')
+    return copy.deepcopy(_atom_data)
```
Comment (yeganer) [inline tardis/conftest.py]: As `copy.deepcopy` doesn't seem to work here we might as well get rid of `atom_data` and rename `_atom_data` to `atom_data`

File: tardis/simulation/tests/test_simulation.py [hunk 3]
```diff
@@ -2,16 +2,18 @@
...
+from astropy.tests.helper import assert_quantity_allclose, remote_data
+
 
 @pytest.fixture
-def tardis_config(kurucz_atomic_data, tardis_config_verysimple):
-    return config_reader.Configuration.from_config_dict(
-        tardis_config_verysimple, atom_data=kurucz_atomic_data)
+def tardis_config(atom_data, tardis_config_verysimple):
+    return Configuration.from_config_dict(
+        tardis_config_verysimple, atom_data=atom_data
```
Comment (yeganer) [inline tardis/simulation/tests/test_simulation.py]: I think we need a copy here

File: tardis/simulation/tests/test_simulation.py [hunk 4]
```diff
@@ -37,6 +39,7 @@ def simulation_compare_data(simulation_compare_data_fname):
     return h5py.File(simulation_compare_data_fname, mode='r')
 
 
+@remote_data
```
Comment (yeganer) [inline tardis/simulation/tests/test_simulation.py]: I think `@remote_data` for the `atom_data` fixture should be enough, or is there a problem?

File: tardis/tests/integration_tests/test_integration.py [hunk 5]
```diff
@@ -1,15 +1,17 @@
...
-from astropy.tests.helper import assert_quantity_allclose
+from astropy.tests.helper import assert_quantity_allclose, remote_data
 
-from tardis.atomic import AtomData
 from tardis.simulation.base import run_radial1d
 from tardis.model import Radial1DModel
 from tardis.io.config_reader import Configuration
 
 
+@remote_data
```
Comment (yeganer) [inline tardis/tests/integration_tests/test_integration.py]: again I think we need `@remote_data` only for the top level fixture

File: tardis/tests/test_atomic.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import os
 import pytest
 from numpy import testing
+from astropy.tests.helper import remote_data
```
Comment (yeganer) [inline tardis/tests/test_atomic.py]: Same as other comments about `@remote_data`

File: tardis/tests/integration_tests/test_integration.py (hunk 1 above)
Comment (karandesai-96) [inline tardis/tests/integration_tests/test_integration.py]: `run_tardis` was proposed way back earlier but I exactly cant remember why it was turned down. For the first comment, it was to simply check the name of atom data but we dont need it yet so I am removing it.

File: tardis/simulation/tests/test_simulation.py (hunk 4 above)
Comment (karandesai-96) [inline tardis/simulation/tests/test_simulation.py]: There is a problem.

File: tardis/tests/integration_tests/test_integration.py (hunk 5 above)
Comment (karandesai-96) [inline tardis/tests/integration_tests/test_integration.py]: Yes, problem here as well.

File: tardis/tests/integration_tests/test_integration.py (hunk 1 above)
Comment (karandesai-96) [inline tardis/tests/integration_tests/test_integration.py]: Oh this diff is outdated one.
